applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClientBuilder.java

```diff
@@ -26,7 +26,30 @@
 import java.util.Objects;
 
 /**
- * Provides configuration options for instances of {@link ConfigurationAsyncClient}.
+ * Builds instances of {@link ConfigurationAsyncClient} based on the configuration options stored in the builder when
+ * {@link ConfigurationAsyncClientBuilder#build() build} is called. Preparing the builder has two primary options, using
+ * {@link ConfigurationClientCredentials} or using a {@link HttpPipeline} and the service endpoint.
+ *
+ * <p>ConfigurationClientCredentials has the service endpoint and authorization information built into it. Use
+ * {@link ConfigurationAsyncClientBuilder#credentials(ConfigurationClientCredentials) credentials} to configure the builder.</p>
```

Comment:
Let's sync up about these first two paragraphs. I think there needs to be a few tweaks. Notably:

1. 'Builds' is not a good start to the paragraph. We should say something like "This class provides a fluent builder API to help aid the configuration and instantiation of the ...."
2. I don't like the "Preparing the builder has two primary options" sentence. It is hard to parse, and too detailed for a first paragraph.
3. I think if you put yourself in the shoes of a newcomer to this API, the second paragraph above would provide far too little actionable information.